src/rt/sched/backpressure.h

```diff
@@ -0,0 +1,249 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+/**
+ * ## Introduction
+ *
+ * The primary goal of backpressure is to prevent runaway growth of a cown's
+ * message queue that might eventually result in a Verona application running
+ * out of memory. Cowns that receive messages at a higher rate than they are
+ * able to process are considered "overloaded" and may lead to this unbounded
+ * queue growth. Once an overloaded cown is identified, cowns sending messages
+ * to it will be temporarily descheduled ("muted") so that the overloaded cown
+ * can work through its queue. Similarly, cowns sending to muted receivers will
+ * also be muted so that cowns do not themselves become overloaded as a
+ * consequence of being muted. Muted cowns will be rescheduled after the
+ * receiver that triggered their muting (the "mutor") is no longer muted or
+ * overloaded.
+ *
+ * ## Detecting Overloaded Cowns
+ *
+ * An overloaded cown is one that experiences message queue growth as it is
+ * processing messages. Since it would be prohibitively expensive to calculate
+ * the length of a message queue directly, a count of messages removed from the
+ * queue over some period is used as an approximation for message queue size.
+ * Like scheduler fairness, the period is determined using a token message that
+ * will occasionally fall out of the queue and then be inserted to begin the
+ * next period. The "load" of a cown at any time is the accumulation of messages
+ * processed over multiple periods.
+ *
+ * ## Muting Cowns
+ *
+ * All messages from one set of sending cowns to a set of receiving cowns are
+ * scanned to determine if the senders should be muted. The senders will be
+ * muted once their action completes if all of the following are true:
+ *   1. The receiver set does not contain any of the senders.
```

Comment:
I'm not sure if it should be detailed here, but I'm having a really hard time imagining what the scenario is where a cown is in both the sending and receiving sets.

Reply:
This would happen on a recursive behaviour call or some form of continuation for a set of cowns.

Reply:
That definitely sounds like something that shouldn't be detailed right here as it would derail from the key point. However, I do think that going "a level deeper" would be helpful for people in the future, however, I don't see that as required for now. 

If you agree that additional deeper details would be valuable to help others work with the code in the future (I think it would) then I think an issue should be created (assigned to you) to address that.

Given you are far and away the best person to write that, I think we would want to make it enough of a priority to complete before your contract ends.

I imagine the "one level deeper" additions to be the sort of information like the above that you would give someone if you were handing off maintenance of this to them.

Information something like "great know you understand in general how it works, here's some nitty gritty things you need to know to understand why some things are the way they are".